src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,97 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Post } from '../types/Post';
+
+type Props = {
+  addComment: (
+    value1: string,
+    value2: string,
+    value3: string,
+  ) => void,
+  selectedPost: Post | null,
+  setIsButtonNotVisible: (value: boolean) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addComment,
+  selectedPost,
+  setIsButtonNotVisible,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [hasNameError, setHasNameError] = useState(false);
+  const [hasEmailError, setHasEmailError] = useState(false);
+  const [hasCommentTextError, setHasCommentTextError] = useState(false);
+  const [isCommentLoading, setIsCommentLoading] = useState(false);
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!name.trim().length) {
+      setHasNameError(true);
+    }
+
+    if (!email.trim().length) {
+      setHasEmailError(true);
+    }
+
+    if (!commentText.length) {
+      setHasCommentTextError(true);
+    }
+
+    if (name.trim().length && email.trim().length && commentText) {
+      addComment(name, email, commentText);
+      setIsCommentLoading(false);
+      setCommentText('');
+    }
+  };
+
+  const nameHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
```suggestion
  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364